architecture/adr-list/adr-007-estimating-transaction-fees.md

```diff
@@ -0,0 +1,133 @@
+
+> * **Make sure blocks are not consuming too many resources and will be finalized**. This is implemented by default in the SDK via the [block gas meter](https://docs.cosmos.network/v0.43/basics/gas-fees.html#block-gas-meter).
+> * **Prevent spam and abuse from end-user**. To this end, `gas` consumed during [`message`](https://docs.cosmos.network/v0.43/building-modules/messages-and-queries.html#messages) execution is typically priced, resulting in a `fee` \(`fees = gas * gas-prices`\). `fees` generally have to be paid by the sender of the `message`. Note that the SDK does not enforce `gas` pricing by default, as there may be other ways to prevent spam \(e.g. bandwidth schemes\). Still, most applications will implement `fee` mechanisms to prevent spam. This is done via the [`AnteHandler`](https://docs.cosmos.network/v0.43/basics/gas-fees.html#antehandler).
+
+### Investigations related to `gas` and `fees` parameters
+
+#### `gas-wanted`
+
+`gas-wanted` is the amount of gas that a transaction is allowed to use and is set by the transaction _sender_.
+
+While there is an estimation functionality in Cosmos CLI to predict what the gas needed for a transaction should be, it doesn't take into account reads and writes to state. As a consequence, it can be inaccurate and result in a scenario where `gas-wanted` is underestimated.
+
+The `gas-wanted` estimation functionality can also be accessed via RPC calls by querying the `app/simulate` endpoint.
+
+#### **`gas-used`**
+
+`gas-used` is the amount of gas used by a transaction and is always less than  `gas-wanted`.
+
+If a transaction requires more gas than specified in `gas-wanted`, execution is interrupted. The fee is NOT returned and becomes fully exhausted.
+
+#### **`gas-prices`**
+
+`gas-prices` is the price of gas suggested by transaction sender. It is set by the transaction sender and can be set to any value.
+
+Validators can define a parameter called `min-gas-prices`, which acts as a filter on the transactions in each validators' mempool. Transactions with lower `gas-prices` are not considered as candidates for inclusion into the next blocks by the validator. This value is configured by the node/validator operator in the `app.toml` config file.
+
+If `gas-prices` specified in a transaction is lower than the `min-gas-prices` of _all_ validators, no validators consider that transaction in their mempool. The transaction eventually times out and is not processed.
+
+If `gas_prices` specified in a transaction is lower than `min-gas-prices` of _some_ validators, the transaction has a chance at being committed within the transaction timeout window. Whether this actually happens depends on the sum of voting powers of validators that are ready to commit it.
+
+There is currently no way to request `min-gas-prices`.
```

Comment:
The original text said "there is no way to request gas-prices". Since gas-prices are set by the transaction sender, and dynamically at run-time, I don't think they can be "queried". Am I correct in thinking that what you actually meant to say is there's no way to query min-gas-prices, which is a static (as long as it's in the config file) per validator?

Reply:
Correct, my bad wording here